File: Java/BOJ/12018 Yoensi TOTO.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        List<Integer> lecture = new ArrayList<>();
        
        for(int k = 0; k < N; k++) {
        	st = new StringTokenizer(br.readLine());
        	int P = Integer.parseInt(st.nextToken());
        	int L = Integer.parseInt(st.nextToken());
        	List<Integer> m = new ArrayList<>();
        	st = new StringTokenizer(br.readLine());
        	for (int i = 0 ; i < P; i++) {
        		m.add(Integer.parseInt(st.nextToken()));
        	}
        	
        	Collections.sort(m, Collections.reverseOrder());
        	if ( P < L) {
        		lecture.add(1);
        	}
        	else {
        		lecture.add(m.get(L-1));
        	}
        }
        Collections.sort(lecture);
        int result =0;
        for (int i : lecture) {
        	if(M >= i) {
        		result += 1;
        		M -= i;
        	}
        }
        System.out.println(result);
    }
}

File: Python/BOJ/12018 Yoensi TOTO.py
import sys

N, M = map(int, sys.stdin.readline().split())
lecture = []
for k in range(N):
    P, L = map(int, sys.stdin.readline().split())
    m = sorted(list(map(int, sys.stdin.readline().split())), reverse=True)
    if P < L:
        lecture.append(1)
    else:
        lecture.append(m[L - 1])

lecture.sort()
result = 0
for i in lecture:
    if M >= i:
        result += 1
        M -= i
print(result)
